`rptest`: assert `!_stopped` in `KgoVerifierServices::wait()` implementations

WillemKauf opened the pull request:
`KgoVerifierService` objects should be waited upon and then stopped, not the other way around.

## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

* none

WillemKauf committed:
`rptest`: assert `!_stopped` in `wait()` implementations

`KgoVerifierService` objects should be waited upon and then stopped,
not the other way around.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/71026-->
<!--5a6c16bc-ca2a-416e-860c-31715fbbc800-->
<!--7a13015e-0836-4ac4-9a21-e794e6848860-->
<!--599014c3-8051-48af-9944-c1285b9d92bb-->
<!--9c1381f1-224f-46f3-ae16-d87cfce32e57-->
<!--f75cccbd-bc3f-4242-8e5a-8a98351773ea-->
<!--7b1fcc58-1c46-480f-a9fb-199744b20b0e-->
<!--1fb5a227-f890-4c56-8a61-0075c121bfe1-->
<!--3a62f166-ffb5-4e9c-8691-16c1088b05cd-->
<!--3042cae1-219b-42ed-8e8c-a557dcc29465-->
<!--2a9d4d28-7682-4236-827f-7aceb4dfc386-->
<!--a4941f0a-23b3-44a4-8935-2ea192e90b5e-->
<!--781f010e-fa60-44ca-a9a5-8459bbec587a-->
<!--84469295-3060-42c7-aef3-7912eed7c9d6-->
<!--3998eff3-07c9-44f1-ae93-90ebc5b31442-->
<!--title-->
## Retry command for [Build#71026](https://buildkite.com/redpanda/redpanda/builds/71026)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_prefetch_chunks@{"prefetch":0}
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_fallback_mode
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_read_when_cache_smaller_than_segment_size
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_read_chunks
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_prefetch_chunks@{"prefetch":5}
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_read_when_segment_size_smaller_than_chunk_size
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_prefetch_chunks@{"prefetch":3}
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_read_when_chunk_api_disabled
tests/rptest/tests/e2e_shadow_indexing_test.py::ShadowIndexingManyPartitionsTest.test_many_partitions_shutdown
tests/rptest/tests/nodes_decommissioning_test.py::NodesDecommissioningTest.test_learner_gap_metrics
```
<!--/body-->

WillemKauf committed:
`rptest`: correct `stop/wait` call order in `many_partitions_test`

WillemKauf committed:
`rptest`: correct `stop/wait` call order in `cloud_storage_chunk_read_path_test`

WillemKauf committed:
`rptest`: correct `stop/wait` call order in `e2e_shadow_indexing_test`

WillemKauf committed:
`rptest`: correct `stop/wait` call order in `workload_producer_consumer`

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/71031-->
<!--202c6204-cbf3-4959-b0ce-4030a61176d4-->
<!--ed0b64e8-3de8-4885-abb7-3e4c80578c5e-->
<!--3deaecf9-1320-425f-8b5a-98daaef011ea-->
<!--0428c8da-6e7d-4f76-970b-0541e09d1669-->
<!--22938b1a-6711-4544-9d1f-0ef31454a086-->
<!--1209c6d1-f617-452e-8969-9178b5ac6502-->
<!--title-->
## Retry command for [Build#71031](https://buildkite.com/redpanda/redpanda/builds/71031)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_read_when_chunk_api_disabled
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_fallback_mode
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_read_when_segment_size_smaller_than_chunk_size
tests/rptest/tests/cloud_storage_chunk_read_path_test.py::CloudStorageChunkReadTest.test_read_chunks
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/71034-->
<!--3cf1819a-a88c-4b58-a7d9-72d9c89c5569-->
<!--3628d2ef-7c1a-42c1-8cb9-260cc6be6f22-->
<!--title-->
## Retry command for [Build#71034](https://buildkite.com/redpanda/redpanda/builds/71034)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/nodes_decommissioning_test.py::NodesDecommissioningTest.test_learner_gap_metrics
tests/rptest/tests/e2e_shadow_indexing_test.py::ShadowIndexingTrafficShapingTest.test_many_partitions_recovery_with_traffic_shaping
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--e7d5959e-2862-45ee-b01b-042f18799370-->
<!--TEST RESULTS-->
<!--4a464741-634f-4eaa-a0c9-a007f56cc955-->
<!--TEST RESULTS-->
<!--dad9a515-503f-4ff0-ac90-ae6bfec248c9-->
<!--TEST RESULTS-->
<!--7746747a-29a5-4f1d-b47b-39caf371da17-->
<!--TEST RESULTS-->
<!--f2a6f198-ff16-4ab0-acba-c21769ab31b8-->
<!--TEST RESULTS-->
<!--976f8e25-70ce-4906-9a59-e304649f78a5-->
<!--TEST RESULTS-->
<!--d18e7979-aa1c-4074-bb07-a3604a2150eb-->
<!--TEST RESULTS-->
<!--967546fe-8127-4a66-b0b8-4c8131f134ed-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71034">build#71034</a></summary>


| test_class                       | test_method                                        | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 |
|:---------------------------------|:---------------------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|
| ShadowIndexingManyPartitionsTest | test_many_partitions_shutdown                      | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71034#0198c558-0e9b-4543-beb8-d242710c5528 | FAIL          | 0/21     |                                        |
| ShadowIndexingTrafficShapingTest | test_many_partitions_recovery_with_traffic_shaping | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71034#0198c558-0e9c-4909-8ef4-7d47ec51d546 | FAIL          | 0/21     | The test has failed across all retries |
| NodesDecommissioningTest         | test_learner_gap_metrics                           | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71034#0198c558-0e99-4579-9cb1-615d5bddffdd | FAIL          | 0/21     | The test has failed across all retries |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71074">build#71074</a></summary>


| test_class                       | test_method                   | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:---------------------------------|:------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTxTest        | test_write                    | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71074#0198c7e5-fa35-4a98-86a5-7126d3568fe0 | FLAKY         | 18/21    | upstream reliability is '93.12883435582822'. current run reliability is '85.71428571428571'. drift is 7.41455 and the allowed drift is set to 50. The test should PASS |
| ShadowIndexingManyPartitionsTest | test_many_partitions_recovery | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71074#0198c7df-8c69-4d32-b0c3-62ee1b0428b2 | FAIL          | 0/5      |                                                                                                                                                                        |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71084">build#71084</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                     |
|:-------------------------|:---------------------|:----------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 2, "compaction_mode": "adjacent_merge", "enable_failures": false, "mixed_versions": true, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/71084#0198c883-c5a8-4af9-a7fd-ad391a0e380b | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71154">build#71154</a></summary>


| test_class                    | test_method              | test_arguments                                                                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 |
|:------------------------------|:-------------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|
| FeaturesMultiNodeUpgradeTest  | test_rollback            | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-0823-4526-a451-83bd15e4923a | FAIL          | 0/21     | The test has failed across all retries |
| FeaturesMultiNodeUpgradeTest  | test_rollback            | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8544-431b-8660-3fc942108d0e | FAIL          | 0/21     | The test has failed across all retries |
| FeaturesMultiNodeUpgradeTest  | test_upgrade             | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-0824-4d47-9d34-97d500413903 | FAIL          | 0/21     | The test has failed across all retries |
| FeaturesMultiNodeUpgradeTest  | test_upgrade             | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8545-4fff-9b6d-b22f0e73b8ed | FAIL          | 0/21     | The test has failed across all retries |
| FeaturesNodeJoinTest          | test_old_node_join       | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-0825-4f4a-be4e-1e3566b14cef | FAIL          | 0/21     | The test has failed across all retries |
| FeaturesNodeJoinTest          | test_old_node_join       | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8545-4afd-8140-5fca75b99477 | FAIL          | 0/21     | The test has failed across all retries |
| FeaturesSingleNodeUpgradeTest | test_upgrade             | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-081f-4c3a-983a-7bfd271e0692 | FAIL          | 0/1      |                                        |
| FeaturesSingleNodeUpgradeTest | test_upgrade             | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8540-472a-9566-369415bba173 | FAIL          | 0/21     | The test has failed across all retries |
| LicenseEnforcementTest        | test_license_enforcement | {"clean_node_after_recovery": false, "clean_node_before_recovery": false}                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-081d-4b15-86c1-695f334fea11 | FAIL          | 0/21     | The test has failed across all retries |
| LicenseEnforcementTest        | test_license_enforcement | {"clean_node_after_recovery": false, "clean_node_before_recovery": false}                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8543-4ec1-872b-b00dae967717 | FAIL          | 0/21     | The test has failed across all retries |
| LicenseEnforcementTest        | test_license_enforcement | {"clean_node_after_recovery": true, "clean_node_before_recovery": false}                                                               | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-081e-480b-893a-a2dc86c6d34b | FAIL          | 0/21     | The test has failed across all retries |
| LicenseEnforcementTest        | test_license_enforcement | {"clean_node_after_recovery": true, "clean_node_before_recovery": false}                                                               | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8544-431b-8660-3fc942108d0e | FAIL          | 0/21     | The test has failed across all retries |
| LicenseEnforcementTest        | test_license_enforcement | {"clean_node_after_recovery": false, "clean_node_before_recovery": true}                                                               | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-081f-4c3a-983a-7bfd271e0692 | FAIL          | 0/1      |                                        |
| LicenseEnforcementTest        | test_license_enforcement | {"clean_node_after_recovery": false, "clean_node_before_recovery": true}                                                               | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8545-4fff-9b6d-b22f0e73b8ed | FAIL          | 0/21     | The test has failed across all retries |
| LicenseEnforcementTest        | test_license_enforcement | {"clean_node_after_recovery": true, "clean_node_before_recovery": true}                                                                | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-0820-42b6-bd65-4134a31211a1 | FAIL          | 0/21     | The test has failed across all retries |
| LicenseEnforcementTest        | test_license_enforcement | {"clean_node_after_recovery": true, "clean_node_before_recovery": true}                                                                | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8545-4afd-8140-5fca75b99477 | FAIL          | 0/21     | The test has failed across all retries |
| RandomNodeOperationsTest      | test_node_operations     | {"cloud_storage_type": 2, "compaction_mode": "adjacent_merge", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-081f-4c3a-983a-7bfd271e0692 | FAIL          | 0/1      |                                        |
| TopicIDUpgradeTest            | test_topic_id_migration  | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccef-0821-48cd-93fe-ea5ccf12fe1d | FAIL          | 0/21     | The test has failed across all retries |
| TopicIDUpgradeTest            | test_topic_id_migration  | null                                                                                                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/71154#0198ccf5-8547-44dc-9607-81ff1befbf94 | FAIL          | 0/21     | The test has failed across all retries |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71191">build#71191</a></summary>


| test_class               | test_method              | test_arguments                                                                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                               |
|:-------------------------|:-------------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeE2ETests         | test_json_schema_unicode | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "query_engine": "trino"}                                                      | integration | https://buildkite.com/redpanda/redpanda/builds/71191#0198ce68-c7aa-45a0-ac87-abbd5941f7ec | FLAKY         | 20/21    | upstream reliability is '98.37209302325581'. current run reliability is '95.23809523809523'. drift is 3.134 and the allowed drift is set to 50. The test should PASS |
| RandomNodeOperationsTest | test_node_operations     | {"cloud_storage_type": 2, "compaction_mode": "sliding_window", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/71191#0198ce68-c7aa-45a0-ac87-abbd5941f7ec | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS            |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71239">build#71239</a></summary>


| test_class                       | test_method                | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    |
|:---------------------------------|:---------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| NodesDecommissioningTest         | test_learner_gap_metrics   | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71239#0198d213-ad64-4483-89a6-bed5a8618976 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                    |
| WriteCachingFailureInjectionTest | test_unavoidable_data_loss | null             | integration | https://buildkite.com/redpanda/redpanda/builds/71239#0198d213-ad6b-497b-9caa-32f8cf746895 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71250">build#71250</a></summary>


| test_class                | test_method              | test_arguments                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:--------------------------|:-------------------------|:----------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTxTest | test_write               | null                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/71250#0198d2a8-5fc9-4f6d-b37b-04ce81e20388 | FLAKY         | 19/21    | upstream reliability is '92.6056338028169'. current run reliability is '90.47619047619048'. drift is 2.12944 and the allowed drift is set to 50. The test should PASS  |
| DatalakeE2ETests          | test_json_schema_unicode | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "query_engine": "trino"} | integration | https://buildkite.com/redpanda/redpanda/builds/71250#0198d2af-325f-458e-9d4f-c355179eda5f | FLAKY         | 20/21    | upstream reliability is '98.39080459770115'. current run reliability is '95.23809523809523'. drift is 3.15271 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72609">build#72609</a></summary>


| test_class              | test_method                                                     | test_arguments                                                                                                                        | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                   | test_history                                                                                                                                                                                       |
|:------------------------|:----------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ClusterRateQuotaTest    | test_client_group_consume_rate_throttle_mechanism               | null                                                                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/72609#019963a6-197a-43b8-be18-c698ba17705a | FLAKY         | 16/21    | upstream reliability is '98.42931937172776'. current run reliability is '76.19047619047619'. drift is 22.23884 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterRateQuotaTest&test_method=test_client_group_consume_rate_throttle_mechanism               |
| ClusterRateQuotaTest    | test_client_group_produce_rate_throttle_mechanism               | null                                                                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/72609#019963ac-2d3d-4fc1-a57f-3a4a6e23a22d | FLAKY         | 13/21    | upstream reliability is '100.0'. current run reliability is '61.904761904761905'. drift is 38.09524 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterRateQuotaTest&test_method=test_client_group_produce_rate_throttle_mechanism               |
| ClusterRateQuotaTest    | test_client_response_and_produce_throttle_mechanism             | null                                                                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/72609#019963a6-1973-4537-8bdf-9a1dde2fdf78 | FLAKY         | 16/21    | upstream reliability is '98.69451697127938'. current run reliability is '76.19047619047619'. drift is 22.50404 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterRateQuotaTest&test_method=test_client_response_and_produce_throttle_mechanism             |
| ClusterRateQuotaTest    | test_client_response_throttle_mechanism_applies_to_next_request | null                                                                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/72609#019963a6-1975-4447-b2fb-cf8339d38504 | FLAKY         | 15/21    | upstream reliability is '100.0'. current run reliability is '71.42857142857143'. drift is 28.57143 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterRateQuotaTest&test_method=test_client_response_throttle_mechanism_applies_to_next_request |
| SchemaEvolutionE2ETests | test_legal_schema_evolution                                     | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "produce_mode": "avro", "query_engine": "trino", "test_case": "drop_column"} | integration | https://buildkite.com/redpanda/redpanda/builds/72609#019963ac-2d41-43e2-abdc-66b9648dfac3 | FLAKY         | 20/21    | upstream reliability is '99.74025974025975'. current run reliability is '95.23809523809523'. drift is 4.50216 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaEvolutionE2ETests&test_method=test_legal_schema_evolution                                  |
| NodeWiseRecoveryTest    | test_recovery_local_data_missing                                | {"wait_for_final_manifest_uploads": false}                                                                                            | integration | https://buildkite.com/redpanda/redpanda/builds/72609#019963ac-2d42-488e-aa92-ee2b6bb6253d | FLAKY         | 16/21    | upstream reliability is '100.0'. current run reliability is '76.19047619047619'. drift is 23.80952 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodeWiseRecoveryTest&test_method=test_recovery_local_data_missing                                |
| DisablingPartitionsTest | test_disable                                                    | null                                                                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/72609#019963a6-1973-4537-8bdf-9a1dde2fdf78 | FLAKY         | 13/21    | upstream reliability is '98.74055415617129'. current run reliability is '61.904761904761905'. drift is 36.83579 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DisablingPartitionsTest&test_method=test_disable                                                 |
</details>
<!--/body-->

WillemKauf commented:
/ci-repeat 1
release
skip-redpanda-build
skip-units
dt-repeat=1
tests/rptest/scale_tests/ht_partition_movement_test.py

WillemKauf commented:
/ci-repeat 1
release
skip-redpanda-build
skip-units
dt-repeat=1
tests/rptest/scale_tests/many_partitions_test.py

WillemKauf commented:
/ci-repeat 1
release
skip-redpanda-build
skip-units
dt-repeat=1
tests/rptest/scale_tests/partition_balancer_scale_test.py

WillemKauf commented:
/ci-repeat 1
release
skip-redpanda-build
skip-units
dt-repeat=1
tests/rptest/scale_tests/tiered_storage_io_stress_test.py

nvartolomei commented on tests/rptest/services/kgo_verifier_services.py:
```diff
@@ -282,12 +282,7 @@ def wait_node(self, node, timeout_sec=None):
-                f"Can't wait {self.who_am_i()}. It was already stopped."
-                f" You can either stop() a service or wait() and then stop() it"
-                f" but not the other way around.")
+        assert not self._stopped, f"Can't wait {self.who_am_i()}. It was already stopped. You can either stop() a service or wait() and then stop() it but not the other way around."
```
can probably contribute this to ducktape itself

nvartolomei commented on tests/rptest/scale_tests/ht_partition_movement_test.py:
```diff
@@ -100,6 +105,7 @@ def test_moving_single_partition_under_load(self, replication_factor):
             self._do_move_and_verify(t, partition, 600)
 
         self.verify(topic.name)
+        self.cleanup()
```
verify already called wait(). cleanup is not necessarily needed then, right?

WillemKauf replied:
Removed

nvartolomei commented on tests/rptest/scale_tests/partition_balancer_scale_test.py:
```diff
@@ -70,12 +69,17 @@ def _start_consumer(self, topic_name, msg_size, consumers):
         # Await the consumer that is reading only the subset of data that
         # was written before it started.
         self.consumer.wait()
+        self.consumer.stop()
```
this should be a noop, right? wait encapsulates a stop

WillemKauf replied:
Removed

nvartolomei commented on tests/rptest/scale_tests/tiered_storage_io_stress_test.py:
```diff
@@ -169,3 +174,4 @@ def test_io_stress(self, cloud_storage_type, segment_size,
         self._create_initial_topics()
 
         self._workload(segment_size, interval_uploads)
+        self.cleanup()
```
ditto: why do we need this? it is all dead code to me

WillemKauf replied:
Removed

nvartolomei reviewed (dismissed):
mostly questions about dead code; the assert is useful to prevent incorrect use/assumptions

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/71154-->
<!--38cec12d-d0f6-481e-98ca-70c3b4625a42-->
<!--03c50c7d-9073-4e9b-9582-98e3b637f862-->
<!--c38ea5c2-9d28-41ba-b92b-7e9bd4e59f71-->
<!--53bb22c9-df5a-40a7-95e0-5267fdfa9fb5-->
<!--8cf8d861-db23-4242-9896-e8cb92c2437a-->
<!--7edc1ebb-43eb-4fb6-8848-e12ff9393828-->
<!--4f526515-5a2a-4fe6-a965-c94b000dac1c-->
<!--8d74bc95-b532-45a4-b809-3bcefef230f9-->
<!--a2b81ce8-2f84-42af-8c23-b388bbb519bc-->
<!--88c47303-4418-42d2-92bf-2eb40e810e3c-->
<!--360f2cd0-1880-405f-8bcc-3cd01af3b71b-->
<!--7253fd86-57b6-43a0-8b39-c2f525c2783a-->
<!--ec8331b8-7300-484a-972d-0b7b43163d7d-->
<!--391af6f2-57ae-4015-a12f-574486f6d773-->
<!--dc75b3cc-068b-4372-99bf-060948974ba1-->
<!--ee49b517-cd8e-4afb-a049-4a4afe931fcb-->
<!--title-->
## Retry command for [Build#71154](https://buildkite.com/redpanda/redpanda/builds/71154)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_features_test.py::FeaturesSingleNodeUpgradeTest.test_upgrade
tests/rptest/tests/license_enforcement_test.py::LicenseEnforcementTest.test_license_enforcement@{"clean_node_after_recovery":false,"clean_node_before_recovery":false}
tests/rptest/tests/license_enforcement_test.py::LicenseEnforcementTest.test_license_enforcement@{"clean_node_after_recovery":true,"clean_node_before_recovery":false}
tests/rptest/tests/cluster_features_test.py::FeaturesMultiNodeUpgradeTest.test_rollback
tests/rptest/tests/cluster_features_test.py::FeaturesNodeJoinTest.test_old_node_join
tests/rptest/tests/license_enforcement_test.py::LicenseEnforcementTest.test_license_enforcement@{"clean_node_after_recovery":true,"clean_node_before_recovery":true}
tests/rptest/tests/cluster_features_test.py::FeaturesMultiNodeUpgradeTest.test_upgrade
tests/rptest/tests/topic_id_migrator_test.py::TopicIDUpgradeTest.test_topic_id_migration
tests/rptest/tests/license_enforcement_test.py::LicenseEnforcementTest.test_license_enforcement@{"clean_node_after_recovery":false,"clean_node_before_recovery":true}
```
<!--/body-->

WillemKauf commented:
Rebased

nvartolomei commented on tests/rptest/tests/nodes_decommissioning_test.py:
```diff
@@ -490,7 +492,6 @@ def calculate_total_learners_gap() -> int | None:
 
         self.logger.info(f"decommissioning node: {to_decommission_id}", )
         self._decommission(to_decommission_id)
-        self.producer.wait()
```
why is wait removed from there? this changes what the test does no? previously the test waited for consumer to finish before proceeding. but now?

WillemKauf replied:
Oops. reverted.

WillemKauf replied:
>`assert not self._stopped, f"Can't wait {self.who_am_i()}. It was already stopped. You can either stop() a service or wait() and then stop() it but not the other way around."
AssertionError: Can't wait KgoVerifierProducer-0-140548171042240. It was already stopped. You can either stop() a service or wait() and then stop() it but not the other way around.`


Oh, that's why I removed it.

WillemKauf replied:
The second `producer.wait()` call in `verify()` will fail with the new assert if this `wait()` is left here.
It is also unnecessary for the test as written.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/71239-->
<!--6fffaa0b-7c42-4cfc-a475-2a0b12409a96-->
<!--title-->
## Retry command for [Build#71239](https://buildkite.com/redpanda/redpanda/builds/71239)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/nodes_decommissioning_test.py::NodesDecommissioningTest.test_learner_gap_metrics
```
<!--/body-->

WillemKauf committed:
`rptest`: remove `wait()` in `nodes_decommissioning_test`

The second `producer.wait()` call in `verify()` will fail with the new
assert if this `wait()` is left here.

It is also unnecessary for the test as written.

WillemKauf commented:
closing because this isn't the solution to what i thought our problems were

WillemKauf commented:
Reopening at @nvartolomei's request

WillemKauf commented on tests/rptest/scale_tests/many_partitions_test.py:
```diff
@@ -413,7 +413,7 @@ def _restart_stress(
                 err_msg="Waiting for elections to complete after restart",
             )
-            self.logger.info(f"Post-restart elections done.")
+            self.logger.info("Post-restart elections done.")
```
my editor is now applying ruff linting on save FYI (this particular lint removes `f`-strings from strings that don't require it)

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/72169-->
<!--cc2bde41-c3c4-4960-b946-87870091094c-->
<!--title-->
## Retry command for [Build#72169](https://buildkite.com/redpanda/redpanda/builds/72169)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/direct_consumer_tests/direct_consumer_test.py::DirectConsumerVerifierTest.test_basic_consuming_from_topic
```
<!--/body-->

WillemKauf committed:
`rptest`: `direct_consumer_test` fix

WillemKauf commented:
Sorry @nvartolomei I keep forgetting to merge this.

Force push to rebase to `upstream/dev` to fix merge conflicts.

If CI is green by the time you see this, feel free to approve and merge.
